kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/NetConfigManager.java

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.net;
+
+import java.util.Properties;
+
+import org.eclipse.kura.KuraException;
+
+/**
+ * @since 2.0
+ */
+public interface NetConfigManager {
+
+    public void readConfig(NetInterfaceConfig<? extends NetInterfaceAddressConfig> netConfig,
```

Comment:
Add Javadoc explaining what NetInterfaceConfig fields the caller should set before the configuration can be read (probably only the interface name is needed and not modified).

I don't understand the meaning of the second kuraExtendedProps parameter.

I believe that read and write should be enough as method names.

Finally, throwing a generic KuraException is not a good practice unless we document all the possible error codes: the configuration may not be valid, then some IOException may occur...